### ISSUE TITLE
Registering multiple clients at the same time results in duplicate ID's

### ISSUE THREAD

**Issue by vilitux** [wazuh/wazuh#176]
Hey guys, first of all nice work on Wazuh :-).

I've been running into issues yesterday after adding agents on ~50 hosts via Ansible. When trying to register the clients using the auto register script through the API, we've had 8 servers that had a duplicate ID in the end. They didn't seem to error in the first place, as in they were able to register etc, but afterwards they were never able to connect.

The first way to notice that they failed was because of defunct sh processes on the servers that weren't able to connect. Logs also showed that they didn't get a response from the Wazuh server.

Checking via Kibana didn't show any agents not being connected. All seemed healthy enough.
We tried to register the broken agents again, but that resulted in an error message:

```
There is an agent with the same name: stg-srv1
```

Checking on the server itself resulted in this:

```
$ /var/ossec/bin/agent_control -l
Duplicate ID's were like this:
   ID: 037, Name: stg-srv1, IP: 10.10.10.11, Active
   ID: 037, Name: stg-srv2, IP: 10.10.10.12, Never Connected
   ID: 051, Name: srv1, IP: 10.10.10.21, Active
   ID: 051, Name: srv2, IP: 10.10.10.22, Never Connected
```

So there were actually 2 servers with the same ID (which happened 4 times in total) of which 1 of the 2 wasn't able to connect. In order to fix we had to remove both agents and then re-add them. Afterwards everything was fine and there were no more defunct processes.

So basically 2 issues:
- ossec-logcollector spawns child processes but doesn't clean them up if the agent can't connect to the server
- multiple registers via the API allow for duplicate ID's to happen

**Comment by mgmacias95** [comment]
Hi @vilitux,

The registration process of an agent is implemented to avoid the registration of two agents at the same time.

In order to reproduce your problem I need to know: what Wazuh version are you using? And what API version? Are you using `authd` to register the agents?

Best regards,
Marta

**Comment by vilitux** [comment]
Hey Marta,

Thanks for your response.
I've installed the latest versions offered through packages.wazuh.com.

**Wazuh server:**
```
root@wazuh:# apt policy wazuh-api wazuh-manager
wazuh-api:
  Installed: 2.1.0-1xenial
  Candidate: 2.1.0-1xenial
  Version table:
 *** 2.1.0-1xenial 500
        500 https://packages.wazuh.com/apt xenial/main amd64 Packages
        100 /var/lib/dpkg/status

wazuh-manager:
  Installed: 2.1.0-1xenial
  Candidate: 2.1.0-1xenial
  Version table:
 *** 2.1.0-1xenial 500
        500 https://packages.wazuh.com/apt xenial/main amd64 Packages
        100 /var/lib/dpkg/status
```

**Wazuh client:**
```
root@srv1:# apt policy wazuh-agent
wazuh-agent:
  Installed: 2.1.0-1xenial
  Candidate: 2.1.0-1xenial
  Version table:
 *** 2.1.0-1xenial 500
        500 https://packages.wazuh.com/apt xenial/main amd64 Packages
        100 /var/lib/dpkg/status
     2.0.1-1xenial 500
```

Used the following script for registering the agents:
https://raw.githubusercontent.com/wazuh/wazuh-api/2.1/examples/api-register-agent.sh

No `authd` involved in the register process.

**Comment by mgmacias95** [comment]
Hi @vilitux,

This problem is caused because the register via the API (without `authd`) isn't prepared to make multiple registers at the same time. We'll add this option soon but, at the moment, we recommend you to register your agents with `authd` enabled. Once you have [`authd` enabled](https://documentation.wazuh.com/current/user-manual/registering/use-registration-service.html), the registration process will be exactly the same 👍 .

Please tell us if the problem persists using `authd`.

Best regards,
Marta

**Comment by vilitux** [comment]
Hey Marta,

So then it pretty much comes down to this:

```

# (Manager)
openssl req -x509 -batch -nodes -days 365 -newkey rsa:2048 -keyout /var/ossec/etc/sslmanager.key -out /var/ossec/etc/sslmanager.cert

Start the authd server:

# (Manager)
/var/ossec/bin/ossec-authd

Run the auth client on the agent. You must enter the authd server’s IP address, like this:
/var/ossec/bin/agent-auth -m 192.168.1.2
```

I'll give it a try, looks easy enough :-)

**Comment by vilitux** [comment]
Using `authd` there were indeed no issues adding those ~50 hosts again. We'll keep using that for now :+1: .
